src/framework/global/settings.cpp

```diff
@@ -95,6 +95,13 @@ void Settings::reset(bool keepDefaultSettings, bool notifyAboutChanges)
     m_settings->clear();
 
     m_isTransactionStarted = false;
+
+    std::vector<Settings::Key> locallyAddedKeys;
```

Comment:
I made these changes because it turned out that some settings without a value are not present in `m_items`. If the user opens the Preferences dialog, all settings are switched to `m_localSettings`. If the user sets a value of a missing setting, it will be created in `m_localSettings`. Then, if the user resets the preferences without closing the dialog, change notifications will be sent out for the settings in `m_items`, i.e. any settings added in `m_localSettings` will be missed and thefore not updated on the dialog. Two examples of such settings are: **Score -> Style for part** (`PART_STYLE_FILE_PATH` in the code) and **Import -> Style file** (`STYLE_FILE_IMPORT_PATH_KEY` in the code).